stacks_queues/queue.py

```diff
@@ -23,39 +23,88 @@ def enqueue(self, element):
             In the store are occupied
             returns None
         """
-        pass
+        if self.size == self.buffer_size:
+            raise QueueFullException ("Queue is full")
+
+        if self.size == 0:
+            self.front = 0
+            self.rear = 0
+
+        #per the test, don't duplicate the item if it was just enqueued
+        if self.store[(self.rear - 1 ) % self.buffer_size] != element:
```

Comment:
🤔 I'm curious which test caused you to add this clause here. It shouldn't be necessary

Reply:
I went through lots of iterations to debug the tests. This is leftovers lol!